src/gluonts/dataset/pandas.py

```diff
@@ -102,18 +101,25 @@ def __post_init__(self) -> None:
         else:  # case single dataframe
             self._dataframes = [self._to_tuple(self.dataframes)]
 
-        for i, (_, df) in enumerate(self._dataframes):
+        for i, (item_id, df) in enumerate(self._dataframes):
             if self.timestamp:
                 df = df.set_index(keys=self.timestamp)
-                self._dataframes[i] = (self._dataframes[i][0], df)
-            else:
-                assert isinstance(
-                    df.index, DatetimeIndexOpsMixin
-                ), "Please set ``timestamp`` or provide a DatetimeIndex."
+
+            assert isinstance(
+                df.index, DatetimeIndexOpsMixin
+            ), "Please set ``timestamp`` or provide a DatetimeIndex."
```

Comment:
I'm not sure this check is needed, we will just convert the index to `PeriodIndex` below and it's maybe on Pandas to raise an exception